llvm/lib/Transforms/Scalar/LoopStrengthReduce.cpp

```diff
@@ -6808,6 +6808,12 @@ canFoldTermCondOfLoop(Loop *L, ScalarEvolution &SE, DominatorTree &DT,
   if (!matchSimpleRecurrence(LHS, ToFold, ToFoldStart, ToFoldStep))
     return std::nullopt;
 
+  // If ToFold does not have an incoming value from LoopLatch then the simple
+  // recurrence is from a prior loop unreachable from the loop we're currently
+  // considering.
+  if (ToFold->getBasicBlockIndex(LoopLatch) == -1)
```

Comment:
A more idiomatic check here would be: L->getHeader() != ToFold->getParent().  This directly checks the property you're comment indicates.

Reply:
Updated.

Previously I had tried comparing with the LoopLatch which interfered with the SeveralLoopLatch testcase. I'm going to read up on how getHeader differs from LoopLatch. Thanks!